### PR TITLE
Add map controls

### PR DESCRIPTION
**Pull request by jcfilben**
Adds zoom in, zoom out, and home controls to the map.

Screen Shot 2023-06-20 at 10 51 57 AM

### CODE REVIEW

File: src/Map.jsx [hunk 1]
```diff
@@ -12,6 +12,42 @@ const initialStyle = {
   height: "100%",
 };
 
+const ZoomControl = ({ home }) => {
+  const map = useMap();
+  return (
+    <div className="leaflet-bottom leaflet-right">
```
Comment (taysea) [inline src/Map.jsx:18]: Can these be `Box`?

Comment (taysea) [reply src/Map.jsx:18]: Also, for reference -- are you able to add links to the CSS with the values for these classes?

Comment (jcfilben) [reply src/Map.jsx]: Yes, added a comment that links to the leaflet css file

**Review by taysea — COMMENTED** [review]
With the arrow icon we're using, I was expecting that it would center on my location (as opposed to returning to the Denver cluster). This might already be the plan, but is there a different icon we are going to use?

**Comment by jcfilben** [comment]
> With the arrow icon we're using, I was expecting that it would center on my location (as opposed to returning to the Denver cluster). This might already be the plan, but is there a different icon we are going to use?

I agree the arrow icon seems like it should center on your location. I think we might have use cases where we want to center on the users location and other use cases where we would want to center on a cluster. We may need to clarify with the designers if we should be using different icons for these cases.

**Comment by jcfilben** [comment]
Just chatted with Kurt and Vicky. Consensus was to use the Globe icon for zooming to users location and the Target icon for zooming to clusters

File: src/Controls.jsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+              event.preventDefault();
+              map.zoomOut();
+            }}
+          />
+          <Box
+            border={{
+              color: "border-weak",
+              side: "top",
+            }}
+          />
```
Comment (taysea) [inline src/Controls.jsx:56]: Would it make more sense for this Box to contain the button below? soft opinion

File: src/Controls.jsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+          {home ? (
```
Comment (taysea) [inline src/Controls.jsx:57]: Is there a scenario where the user might want a shortcut to both "current location" and "data"? If so, should this be implemented differently?

**Comment by taysea** [comment]
The target makes more sense to me for this context. I'm fine going with that direction for now. Easy to swap out if we choose something else in the future.

File: src/Controls.jsx (hunk 2 above)
Comment (jcfilben) [reply src/Controls.jsx]: Yes, I changed this Box to contain the buttons below in the latest commit

File: src/Controls.jsx (hunk 3 above)
Comment (jcfilben) [reply src/Controls.jsx]: Hmm good thought, I adjusted the code to allow for the flexibility to support this.

**Review by taysea — APPROVED** [review]

File: demo-app/src/LocationsMap.jsx [hunk 4]
```diff
@@ -19,7 +27,14 @@ function LocationsMap() {
   return (
     <Box ref={containerRef} flex background="background-contrast">
       {geolocation && (
-        <Map id="map" ref={mapContainerRef} center={geolocation} zoom={6}>
+        <Map
+          id="map"
+          ref={mapContainerRef}
+          center={LocationBounds({ locations: geojsonLocations }).getCenter()}
```
Comment (halocline) [inline demo-app/src/LocationsMap.jsx:33]: Let's leave this as is for now:

```suggestion
          center={geolocation}
```

File: demo-app/src/LocationsMap.jsx [hunk 5]
```diff
@@ -19,7 +27,14 @@ function LocationsMap() {
...
+          zoom={6}
+          zoomControl={false}
+        >
+          <Controls locations={geojsonLocations} />
```
Comment (halocline) [inline demo-app/src/LocationsMap.jsx:37]: Having this fit the geoJSON bounds makes it difficult to assess the capability here.

Comment (halocline) [reply demo-app/src/LocationsMap.jsx:37]: Perhaps we eliminate the GeoJSON marker cluster and focus on the generated locations?

We may need to move the generateLocations call into a useEffect and store locations in state...

Comment (taysea) [reply demo-app/src/LocationsMap.jsx:37]: Sounds good, let me have a look into this

**Review by halocline — APPROVED** [review]

**Pull request merged**